src/python/pants/backend/python/dependency_inference/scripts/dependency_parser.py

```diff
@@ -79,6 +87,12 @@ def find_token(string):
             if alias.asname and token[1] != alias.asname:
                 find_token(alias.asname)
 
+    def maybe_add_string_dependency(self, node, s):
+        if STRING_IMPORTS and STRING_IMPORT_REGEX.match(s):
+            self.weak_imports.setdefault(s, node.lineno)
+        if ASSETS and ASSET_REGEX.match(s):
+            self.assets.add((None, s))
+
```

Comment:
The only time this is possible is if you set min slashes and min dots to 0, right? Otherwise, this could early return? Given that this regex runs against _every single string_ and regex is relatively fast but does have a perf hit, it might be worth adding that eager exit. Wdyt?

Reply:
(Note the file we're parsing has a bunch of strings)
on `main`:
```
Benchmark 1: STRING_IMPORTS=Y MIN_DOTS=2 python src/python/pants/backend/python/dependency_inference/scripts/import_parser.py src/python/pants/option/global_options.py
  Time (mean ± σ):      33.3 ms ±   1.9 ms    [User: 27.7 ms, System: 5.6 ms]
  Range (min … max):    28.4 ms …  41.1 ms    100 runs
```

with my changes:
```
Benchmark 1: STRING_IMPORTS=Y STRING_IMPORT_MIN_DOTS=2 ASSETS=y ASSET_MIN_SLASHES=1 python src/python/pants/backend/python/dependency_inference/scripts/dependency_parser.py src/python/pants/option/global_options.py
  Time (mean ± σ):      33.4 ms ±   1.7 ms    [User: 28.5 ms, System: 4.9 ms]
  Range (min … max):    28.5 ms …  36.8 ms    100 runs
```

Reply:
Ah okay cool. Nevermind me :) thanks